### PR TITLE
burp: new package (backup program)

### PR DESCRIPTION
**Pull request by tokudan**

### CODE REVIEW

**Comment by domenkozar** [comment]
why keep two versions?

**Comment by tokudan** [comment]
burp has actually three main version:
oldstable, stable and latest
latest is sort of an in progress work and not completely suited for production yet, due to a major rewrite and new protocol and is not feature-complete yet.
stable (1.4.40) is what is currently suggested for new installations, hence it's in default.nix
oldstable is mainly useful if the burp server cannot be upgraded yet, as burp generally supports "new server, older client", but not "newer client, older server". so if the server is still 1.3.48 (e.g. debian jessie), then it's generally considered better to use the matching client (1.3.x) by the author.

File: pkgs/tools/backup/burp/1.3.48.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  patches = [ ./burp_1.3.48.patch ];
+
+  buildInputs = [ librsync_0_9 ncurses openssl zlib ]
+    # next two lines copied from bacula, as burp needs acl as well
+    # acl relies on attr, which I can't get to build on darwin
+    ++ stdenv.lib.optional (!stdenv.isDarwin) acl;
+
+    configureFlags = [
```
Comment (domenkozar) [inline pkgs/tools/backup/burp/1.3.48.nix]: wrong indent

File: pkgs/tools/backup/burp/1.3.48.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  buildInputs = [ librsync_0_9 ncurses openssl zlib ]
+    # next two lines copied from bacula, as burp needs acl as well
+    # acl relies on attr, which I can't get to build on darwin
+    ++ stdenv.lib.optional (!stdenv.isDarwin) acl;
+
+    configureFlags = [ 
+      "--sbindir=$out/bin" 
+    ];
+
+  #installPhase = ''make install'';
```
Comment (domenkozar) [inline pkgs/tools/backup/burp/1.3.48.nix]: remove the line if it's not needed

File: pkgs/tools/backup/burp/1.3.48.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchgit,
+	acl,
+	librsync_0_9,
+	ncurses,
+	openssl,
+	zlib
```
Comment (domenkozar) [inline pkgs/tools/backup/burp/1.3.48.nix]: wrong indent, it has to be one line or begin with a comma (no spaces in front)

**Comment by tokudan** [comment]
found the chapter about the conventions and updated the PR

**Comment by globin** [comment]
@tokudan burp 1.3 is the only thing referencing `librsync_0_9`, is it still needed or feasible to use a newer librsync version?

**Comment by tokudan** [comment]
The burp website still lists 1.3. as old stable, also some distributions still only have that version in their repo, so I think still keeping that version for now makes sense, although I do not know how widespread the usage is. Probably rather low.
If keeping librsync 0.9 causes issues, I think it can be removed.

**Comment by globin** [comment]
Isn't really an issue, just noticed that it was the only package depending on it, while fixing up stuff on it for stdenv hardening and wanted to know the current state of it. :)

**Pull request merged**